README.md

```diff
@@ -199,16 +199,54 @@ that it uses existing drop methods as garbage collection finalisers, saving
 users the potentially error-prone task of manually writing both destructor and
 finaliser methods for GC managed objects.
 
-When a `Gc`'s underlying `GcBox`  becomes unreachable, Alloy will call its
+When a `Gc`'s underlying contents becomes unreachable, Alloy will call its
 finaliser, which means that `drop` is called on all the component types (in the
 same way that Rust automatically calls `drop` in an RAII context).
 
-#### TODO: still need to explain
+To achieve Alloy's goal of making cyclic data structures easier to write, we
+made the decision to support the finalisation of objects with cycles. This means
+that no guarantees are made about the order in which finalisers are run. In
+order for this to be sound, you must not access other `Gc` objects from inside
+the `drop` method a `Gc`. Alloy ensures that this rule is followed by checking
+for potential misuses of `drop` at compile-time in a pass known as _Finaliser
```

Comment:
s/in a pass known as/with/

Reply:
Fixed here: 3855ba4824e745bca5ddb9e1ddc8b75455c3680f